src/util/statistics_registry.h

```diff
@@ -31,6 +31,55 @@
  ** for a longer discussion on symbol visibility.
  **/
 
+
+/**
+ * On the design of the statistics:
```

Comment:
Maybe replicate this documentation in the  corresponding newly introduced `stat_*.(h|cpp)` files? Or move what is relevant?

Reply:
This will be split and moved accordingly with the next PR, where I plan to clean up the ownership issue with the statistics.